### PR TITLE
Ul refactor

### PR DESCRIPTION
**Pull request by kerolasa**
This similar refactoring effort that I did with col(1) back in https://github.com/karelzak/util-linux/pull/1115. While the commits  cause lots of lines to change I do not think these are particularly controversial changes. Or to say the same bit differently, these changes  make ul(1) to join many other util-linux tools that share the idea of control struct, and other ideas that are mostly style things.

### CODE REVIEW

File: text-utils/ul.c [hunk 1]
```diff
@@ -254,12 +239,18 @@ static void sig_handler(int signo __attribute__((__unused__)))
 	_exit(EXIT_SUCCESS);
 }
 
+static int ul_putwchar(int c)
+{
+	if (putwchar(c) == WEOF)
+		return EOF;
```
Comment (karelzak) [inline text-utils/ul.c:232]: That's right. Note that the fallbacks are in include/widechar.h. For example `# define putwchar putchar`.

File: text-utils/ul.c [hunk 2]
```diff
@@ -71,8 +71,8 @@
 #define	FREV	'7'
 
 enum {
-	NORMAL 	= 0,		/* Must be zero, see initbuf() */
-	ALTSET 	= 1 << 0,	/* Reverse */
+	G0_SET 	= 0,		/* Must be zero, see initbuf() */
+	G1_SET 	= 1 << 0,	/* Reverse */
```
Comment (karelzak) [inline text-utils/ul.c:75]: Frankly, the new names are not too readable for me, would be possible to get something more human friendly? ;-)

File: text-utils/ul.c (hunk 1 above)
Comment (kerolasa) [reply text-utils/ul.c:232]: Nice, I had forgot `include/widechar.h` even exists.

File: text-utils/ul.c (hunk 2 above)
Comment (kerolasa) [reply text-utils/ul.c]: I changed that commit to be hopefully better 580c63259bb1b59ec6730a1c4246269d76cb13e2.

**Comment by karelzak** [comment]
thanks!

**Comment by karelzak** [comment]
@kerolasa, It seems travis-ci does not like the change:
  https://travis-ci.org/github/karelzak/util-linux/builds/737683018

```
           ul: basic tests                    ...
diff-{{{
--- /home/travis/build/karelzak/util-linux/tests/expected/ul/basic	2020-10-21 11:36:08.236508999 +0000
+++ /home/travis/build/karelzak/util-linux/tests/output/ul/basic	2020-10-21 11:42:16.524508999 +0000
@@ -103,7 +103,7 @@
 underline
 flush
 afer flush
-normaldim
+normaldim
 normaldimunderlined
 normal
 }}}-diff
 FAILED (ul/basic)
```

Maybe there is different terminal type and tigetstr() returns something else than expected (what about to use `ul --terminal 'something'` in the tests?). Can you check it?

Note that we have also old test in tests/ts/misc/ul, what about to move it to tests/ts/ul?

**Pull request merged**